go/cbor.go

```diff
@@ -475,14 +472,33 @@ func (dec *Decoder) decodeText(rv DecodeValue, cborInfo byte, aux uint64) error
 			}
 		}
 	} else {
-		raw := make([]byte, aux)
-		_, err = io.ReadFull(dec.rin, raw)
+		raw, err := dec.readBytes(aux)
+		if err != nil {
+			return err
+		}
 		xs := string(raw)
 		return rv.SetString(xs)
 	}
 	return errors.New("internal error in decodeText, shouldn't get here")
 }
 
+func (dec *Decoder) readBytes(n uint64) ([]byte, error) {
```

Comment:
I would use a `Buffer` from `bytes` and `LimitedReader` from `io`. You don't have to write it yourself. Just check at the end if length of the Buffer matches how many bytes were supposed to be read.

Reply:
Done + works with go-ipfs